packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_client.dart

```diff
@@ -108,6 +109,14 @@ class EndpointClient {
       await _globalFieldsManager.addMetrics(userProfile.properties!.metrics);
     }
 
+    if (userProfile is AWSPinpointUserProfile) {
+      final attributes = <String, List<String>>{};
+      userProfile.userAttributes.getAllProperties().forEach((key, value) {
+        attributes[key] = [value.toString()];
+      });
+      newUserBuilder.userAttributes = ListMultimapBuilder(attributes);
```

Comment:
```suggestion
      userProfile.userAttributes.getAllProperties().forEach(
            (key, value) => newUserBuilder.userAttributes.add(
              key,
              value.toString(),
            ),
          );
```

Reply:
This brings up a question - should `userAttributes` allow setting multiple values per key? Right now, only one value is allowed but it seems like Pinpoint allows multiple.

Reply:
Great question and catch there.  That is correct Pinpoint allows that. 

Unfortunately Android and iOS only expose it as a single value and not as a list.  This is true for the Endpoint global attributes and for the user attributes as well.  

For consistency I believe we should keep it the same as they do.  That's the general direction I'm aware of from our PM given how we are looking to possibly gut and change all of Analytics anyway in the future ..